reasoner/resolution/resolver/RuleResolver.java

```diff
@@ -21,74 +21,43 @@
 import grakn.common.collection.Either;
 import grakn.common.concurrent.actor.Actor;
 import grakn.core.concept.answer.ConceptMap;
-import grakn.core.reasoner.resolution.framework.Answer;
+import grakn.core.logic.Rule;
 import grakn.core.reasoner.resolution.framework.Request;
-import grakn.core.reasoner.resolution.framework.Resolver;
+import grakn.core.reasoner.resolution.framework.ResolutionAnswer;
 import grakn.core.reasoner.resolution.framework.Response;
 import grakn.core.reasoner.resolution.framework.ResponseProducer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.List;
-
 // TODO unify and materialise in receiveAnswer
 public class RuleResolver extends ConjunctionResolver<RuleResolver> {
     private static final Logger LOG = LoggerFactory.getLogger(RuleResolver.class);
 
-    public RuleResolver(Actor<RuleResolver> self, List<Long> when, Long traversalSize) {
-        super(self, RuleResolver.class.getSimpleName() + "(pattern:" + when + ")", when, traversalSize);
+    public RuleResolver(Actor<RuleResolver> self, Rule rule) {
+        super(self, RuleResolver.class.getSimpleName() + "(rule:" + rule + ")", rule.when(), rule.whenConcludables());
     }
 
     @Override
     public Either<Request, Response> receiveRequest(Request fromUpstream, ResponseProducer responseProducer) {
-        return produceMessage(fromUpstream, responseProducer);
-    }
-
-    @Override
-    public Either<Request, Response> receiveAnswer(Request fromUpstream, Response.Answer fromDownstream, ResponseProducer responseProducer) {
```

Comment:
let's put this back since max will have to implement `materialise()` in here, which is different between the conjunction actors